tzkt/fetchTransactionTree.ts

```diff
@@ -46,85 +52,98 @@ async function fetchTransactions(
   return results;
 }
 
+function isUserWallet(wallet: Wallet) {
+  const address = wallet.address;
+  const isUserWallet = !(
+    wallet.address.startsWith("KT1") ||
+    exchangeAliases.some((alias) => alias === wallet.alias)
+  );
+  console.log(
+    "is Exchange wallet",
+    exchangeAliases.some((alias) => alias === wallet.alias)
+  );
+
+  console.log(`Wallet belongs to a user ${isUserWallet}`, wallet);
+  return isUserWallet;
+}
+
 function sleep(ms) {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
 
-async function fetchUniqueAddresses(
-  addresses: string[],
+async function fetchUniqueWallet(
+  wallets: Wallet[],
   start: Date,
   end: Date,
   direction: Direction.IN | Direction.OUT
-) {
+): Promise<Wallet[]> {
```

Comment:
Pas besoin de preciser le type, le compiler le depuis tout seul.
```suggestion
) {
```